File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        // let's use two pointers
        int pleft = 0;
        int pright = numbers.size() - 1;
        int nbleft = numbers[pleft];
        int nbright = numbers[pright];
        int s;
        vector<int> res;
        while (pleft < pright) {
            s = nbleft + nbright;
            if (s < target) {
                pleft++;
                nbleft = numbers[pleft];
            }
            else if (s > target) {
                pright--;
                nbright = numbers[pright];
            }
            else {
                res.push_back(pleft+1);
                res.push_back(pright+1);
                return res;
            }
        }
        res.push_back(-1);
        res.push_back(-1);
        return res;
    }
};

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        # let's use two pointers
        pleft = 0
        pright = len(numbers) - 1
        nbleft = numbers[pleft]
        nbright = numbers[pright]
        while pleft < pright:
            s = nbleft + nbright
            if s < target:
                pleft += 1
                nbleft = numbers[pleft]
            elif s > target:
                pright -= 1
                nbright = numbers[pright]
            else:
                return [pleft+1, pright+1]
        return [-1, -1]